django/db/models/fields/related_lookups.py

```diff
@@ -120,9 +120,19 @@ def get_prep_lookup(self):
             # doesn't have validation for non-integers, so we must run validation
             # using the target field.
             if self.prepare_rhs and hasattr(self.lhs.output_field, 'path_infos'):
+                from django.db.models.fields.related import ForeignKey
+
                 # Get the target field. We can safely assume there is only one
                 # as we don't get to the direct value branch otherwise.
-                target_field = self.lhs.output_field.path_infos[-1].target_fields[-1]
+                if (
+                    not isinstance(self.lhs.output_field, ForeignKey) and
+                    hasattr(self.lhs.output_field, 'reverse_path_infos')
+                ):
+                    # This must be done when the target field is not a foreign key like with generic relationships.
+                    target_field = self.lhs.output_field.reverse_path_infos[-1].target_fields[-1]
+                else:
+                    target_field = self.lhs.output_field.path_infos[-1].target_fields[-1]
```

Comment:
I'm not sure if this is a good solution.  For me, the main question is why `self.lhs.output_field` is set to `generic_relations.TIntegration.tags` and not `generic_relations.TIntegration` :thinking: